### PR TITLE
fix(wsimport): small fixes, refactoring and improvements

### PR DESCRIPTION
**Pull request by kallesoranko**
* Setting OSS as component type
* Keep "Suspected" in license name if it is there
* Catch `HttpException` in the rest call 
* Small refactoring

### CODE REVIEW

**Comment by mcjaeger** [comment]
(from discussion)
* component type OSS is set automatically always
* suspected is a prefix of whitesource licesne short name

File: backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLicenseToSw360LicenseTranslator.java [hunk 1]
```diff
@@ -15,29 +15,28 @@
...
  */
 public class WsLicenseToSw360LicenseTranslator implements EntityTranslator<WsLicense, org.eclipse.sw360.datahandler.thrift.licenses.License> {
 
     @Override
     public License apply(WsLicense wsLicense) {
-        String license = wsLicense.getName().replaceFirst(SUSPECTED, "").trim();
-        String licenseWithDashes = license.replaceAll("[\\s+/]","-");
+        String licenseName = wsLicense.getName();
+        String licenseShortName = licenseName
+                .replaceAll("[\\s+/]","-");
```
Comment (maxhbr) [inline backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLicenseToSw360LicenseTranslator.java]: Not part of this PR: This would replace `GPL-2.0+` (outdated but still commonly used) with `GPL-2.0-`?

File: backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/rest/WsRestClient.java [hunk 2]
```diff
@@ -53,17 +55,23 @@ private HttpClient getConfiguredHttpClient() {
...
         String input = generateRequestBody(requestString, tokenCredentials.getUserKey(), type, token);
         HttpClient httpClient = getConfiguredHttpClient();
         HttpResponse response = getWsConnection(input, httpClient, tokenCredentials.getServerUrl());
-        return IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+        int statusCode = response.getStatusLine().getStatusCode();
+        if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+            return IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+        } else {
+            LOGGER.info("Request unsuccessful: " + response.getStatusLine().getReasonPhrase());
+            throw new HttpException();
```
Comment (maxhbr) [inline backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/rest/WsRestClient.java]: maybe add also the message to the exception? Especially since it is logged later.

**Comment by mcjaeger** [comment]
@kallesoranko, @maxhbr  please help about the status: shall we remove "needs review" or wait until the comments were answered?

**Comment by kallesoranko** [comment]
@mcjaeger I still need to update pr according to comments

File: backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/rest/WsRestClient.java (hunk 2 above)
Comment (kallesoranko) [reply backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/rest/WsRestClient.java]: Good point. Added log message

File: backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLicenseToSw360LicenseTranslator.java [hunk 3]
```diff
@@ -28,6 +28,10 @@ public License apply(WsLicense wsLicense) {
         String licenseShortName = licenseName
                 .replaceAll("[\\s+/]","-");
 
+        if (licenseShortName.endsWith("-")) {
+            licenseShortName = licenseShortName.substring(0, licenseShortName.length()-1) + "+";
+        }
+
```
Comment (kallesoranko) [inline backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLicenseToSw360LicenseTranslator.java]: Added this maybe "a bit hacky" way to handle GLP-2.0+, I guess there is no license name that ends with "-"?

Comment (maxhbr) [reply backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLicenseToSw360LicenseTranslator.java]: spdx at least does not forbid to have a `-` as trailing character. And yes, this is way to hacky.

Comment (kallesoranko) [reply backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLicenseToSw360LicenseTranslator.java]: removed

File: backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLibraryToSw360ReleaseTranslator.java [hunk 4]
```diff
@@ -37,10 +37,13 @@ public Release apply(WsLibrary wsLibrary) {
         sw360Release.getExternalIds().put(FILENAME, wsLibrary.getFilename());
 
         if (wsLibrary.getReferences() != null) {
-                sw360Release.setDownloadurl(wsLibrary.getReferences().getScmUrl());
+            sw360Release.setDownloadurl(wsLibrary.getReferences().getUrl());
```
Comment (kallesoranko) [inline backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/entitytranslation/WsLibraryToSw360ReleaseTranslator.java]: added this download url for releases and add SCM url as external is

**Comment by maxhbr** [comment]
changed from `needs general test` to `needs special test` since the test needs a whitesource connection.
